### DISCUSSION TITLE
Auto `__all__` generator

### DISCUSSION THREAD

**Discussion by rjkilpatrick** [microsoft/pylance-release#2666] [Enhancement]
When creating an `__init__.py`, it is recommended to create an `__all__` list, explicitly containing the public API at the top of the file in an easy to read manner. It also controls [what is pulled into a namespace](https://docs.python.org/3/tutorial/modules.html#importing-from-a-package), and whilst `from package import *` is not recommended, this at least protects the private API.

I think an automatic generator that parses the active file and generates the `__all__` variable from the classes/functions available, would speed up development time, and encourage good practice.

**Comment by judej** [comment]
@rjkilpatrick, thanks for the suggestion. We will review this.

**Comment by judej** [comment]
Moving this issue to discussion as an enhancement request for comments and upvotes.

**Comment by Linux-cpp-lisp** [comment]
This would be a big timesaver, did it ever get implemented?
